esp-wifi/src/ble/os_adapter_esp32.rs

```diff
@@ -285,26 +287,27 @@ pub(crate) fn create_ble_config() -> esp_bt_controller_config_t {
 
 pub(crate) fn btdm_controller_mem_init() {
     extern "C" {
-        static _data_start_btdm: u32;
+        static mut _data_start_btdm: u32;
+        static mut _data_end_btdm: u32;
         static _data_start_btdm_rom: u32;
-        static _data_end_btdm: u32;
     }
 
     // initialise .data section
     unsafe {
-        let len = (&_data_end_btdm as *const _ as *const u8 as usize)
-            - (&_data_start_btdm as *const _ as *const u8 as usize);
+        let data_start = addr_of_mut!(_data_start_btdm).cast::<u8>();
+        let data_end = addr_of_mut!(_data_end_btdm).cast::<u8>();
 
-        core::ptr::copy_nonoverlapping(
-            _data_start_btdm_rom as *const u8,
-            &_data_start_btdm as *const _ as *mut u8,
-            len,
-        );
+        // It looks weird, but esp-idf converts this value into a pointer, too.
```

Comment:
maybe we could say that `_data_start_btdm_rom` points to the real address in ROM ... something like that
with that information it's not weird anymore I guess

But probably it would be more obvious if they had named this with a `_ptr` suffix like in other places 🤷‍♂️ 
Anyway with some explanation here it should look less weird 👍

Reply:
I'll update it, I added the comment before catching on to the idea.